File: app/src/main/java/com/aplicacion/mypet/activities/publicar/FiltrosActivity.java
package com.aplicacion.mypet.activities.publicar;

import android.os.Bundle;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.aplicacion.mypet.R;
import com.aplicacion.mypet.adaptadores.AdaptadorPublicacion;
import com.aplicacion.mypet.models.Publicacion;
import com.aplicacion.mypet.providers.AuthProvider;
import com.aplicacion.mypet.providers.PublicacionProvider;
import com.aplicacion.mypet.utils.ViewedMessageHelper;
import com.firebase.ui.firestore.FirestoreRecyclerOptions;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.firestore.Query;
import com.google.firebase.firestore.QuerySnapshot;

public class FiltrosActivity extends AppCompatActivity {
    private String extraTipoAnimal;
    private TextView tipoAnimalTextView;
    private TextView contadorAnimalesFiltros;
    private AuthProvider auth;

    private RecyclerView recyclerView;
    private PublicacionProvider publicacionProvider;
    private AdaptadorPublicacion adaptadorPublicacion;

    private LinearLayout linearLayoutNoBusqueda;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_filtros);

        extraTipoAnimal = getIntent().getStringExtra("tipo");

        recyclerView = findViewById(R.id.listar_anuncios_filtro);
        auth = new AuthProvider();


        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(linearLayoutManager);
        publicacionProvider = new PublicacionProvider();

        tipoAnimalTextView = findViewById(R.id.tipo_animal_filtro);
        contadorAnimalesFiltros = findViewById(R.id.contador_animal_filtro);
        linearLayoutNoBusqueda = findViewById(R.id.mensaje_informativo_filtros);

        seleccionartipo();
    }

    @Override
    public void onStart() {
        super.onStart();
        Query query = publicacionProvider.getPublicacionByCategoryAndTimesTamp(extraTipoAnimal);

        query.get().addOnSuccessListener(new OnSuccessListener<QuerySnapshot>() {
            @Override
            public void onSuccess(QuerySnapshot queryDocumentSnapshots) {
                if (queryDocumentSnapshots.size()==0) {
                    linearLayoutNoBusqueda.setVisibility(View.VISIBLE);
                } else {
                    linearLayoutNoBusqueda.setVisibility(View.GONE);
                }
            }
        });

        FirestoreRecyclerOptions<Publicacion> options = new FirestoreRecyclerOptions.Builder<Publicacion>()
                .setQuery(query,Publicacion.class)
                .build();

        if (auth.getAuth().getCurrentUser() != null) {
            ViewedMessageHelper.updateOnline(true, FiltrosActivity.this);
        }

        adaptadorPublicacion = new AdaptadorPublicacion(options,this,contadorAnimalesFiltros);
        recyclerView.setAdapter(adaptadorPublicacion);
        adaptadorPublicacion.startListening();
    }



    @Override
    protected void onPause() {
        super.onPause();
        if (auth.getAuth().getCurrentUser() != null) {
            ViewedMessageHelper.updateOnline(false, this);
        }
    }

    @Override
    public void onStop() {
        super.onStop();
        adaptadorPublicacion.stopListening();
    }

    private void seleccionartipo() {
        String[] tipos = getResources().getStringArray(R.array.lista_animales);
        tipoAnimalTextView.setText(String.format("%s:",tipos[Integer.parseInt(extraTipoAnimal)]));
    }

    public void cerrarFiltro(View view) {
        finish();
    }
}

File: app/src/main/java/com/aplicacion/mypet/activities/publicar/FiltrosActivity.kt
package com.aplicacion.mypet.activities.publicar

import android.os.Bundle
import android.view.View
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.aplicacion.mypet.R
import com.aplicacion.mypet.adaptadores.AdaptadorPublicacion
import com.aplicacion.mypet.models.Publicacion
import com.aplicacion.mypet.providers.AuthProvider
import com.aplicacion.mypet.providers.PublicacionProvider
import com.aplicacion.mypet.utils.ViewedMessageHelper
import com.firebase.ui.firestore.FirestoreRecyclerOptions

class FiltrosActivity : AppCompatActivity() {
    private var extraTipoAnimal: String? = null
    private var tipoAnimalTextView: TextView? = null
    private var contadorAnimalesFiltros: TextView? = null
    private var auth: AuthProvider? = null

    private var recyclerView: RecyclerView? = null
    private var publicacionProvider: PublicacionProvider? = null
    private var adaptadorPublicacion: AdaptadorPublicacion? = null

    private var linearLayoutNoBusqueda: LinearLayout? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_filtros)

        extraTipoAnimal = intent.getStringExtra("tipo")

        recyclerView = findViewById(R.id.listar_anuncios_filtro)
        auth = AuthProvider()


        val linearLayoutManager = LinearLayoutManager(this)
        recyclerView!!.layoutManager = linearLayoutManager
        publicacionProvider = PublicacionProvider()

        tipoAnimalTextView = findViewById(R.id.tipo_animal_filtro)
        contadorAnimalesFiltros = findViewById(R.id.contador_animal_filtro)
        linearLayoutNoBusqueda = findViewById(R.id.mensaje_informativo_filtros)

        seleccionartipo()
    }

    override fun onStart() {
        super.onStart()
        val query = publicacionProvider!!.getPublicacionByCategoryAndTimesTamp(extraTipoAnimal)
        query.get().addOnSuccessListener { queryDocumentSnapshots ->
            if (queryDocumentSnapshots.size() == 0) {
                linearLayoutNoBusqueda!!.visibility = View.VISIBLE
            } else {
                linearLayoutNoBusqueda!!.visibility = View.GONE
            }
        }
        val options = FirestoreRecyclerOptions.Builder<Publicacion>()
                .setQuery(query, Publicacion::class.java)
                .build()
        if (auth!!.auth.currentUser != null) {
            ViewedMessageHelper.updateOnline(true, this@FiltrosActivity)
        }
        adaptadorPublicacion = AdaptadorPublicacion(options, this, contadorAnimalesFiltros)
        recyclerView!!.adapter = adaptadorPublicacion
        adaptadorPublicacion!!.startListening()
    }

    override fun onPause() {
        super.onPause()
        if (auth!!.auth.currentUser != null) {
            ViewedMessageHelper.updateOnline(false, this)
        }
    }

    override fun onStop() {
        super.onStop()
        adaptadorPublicacion!!.stopListening()
    }

    private fun seleccionartipo() {
        val tipos = resources.getStringArray(R.array.lista_animales)
        tipoAnimalTextView!!.text = String.format("%s:", tipos[extraTipoAnimal!!.toInt()])
    }

    fun cerrarFiltro(view: View?) {
        finish()
    }
}